connector/web3/web3.go

```diff
@@ -38,32 +50,85 @@ func (c *web3Connector) Verify(address, msg, signedMsg string) (identity connect
 		return identity, fmt.Errorf("could not decode hex string of signed nonce: %v", err)
 	}
 
+	msgHash := signHash([]byte(msg))
+
 	// This is the v parameter in the signature. Per the yellow paper, 27 means even and 28
 	// means odd.
 	if signb[64] == 27 || signb[64] == 28 {
 		signb[64] -= 27
 	} else if signb[64] != 0 && signb[64] != 1 {
 		// We allow 0 or 1 because some non-conformant devices like Ledger use it.
-		// TODO - Verify using ERC-1271
-		return identity, fmt.Errorf("byte at index 64 of signed message should be 27 or 28: %s", signedMsg)
+		return c.VerifyERC1271Signature(addrb, msgHash, signedMsg)
 	}
 
-	pubKey, err := crypto.SigToPub(signHash([]byte(msg)), signb)
+	pubKey, err := crypto.SigToPub(msgHash, signb)
 	if err != nil {
-		return identity, fmt.Errorf("failed to recover public key from signed message: %v", err)
+		return c.VerifyERC1271Signature(addrb, msgHash, signedMsg)
 	}
 
 	recoveredAddr := crypto.PubkeyToAddress(*pubKey)
 	// These are byte arrays, so this is okay to do.
 	if recoveredAddr != addrb {
-		return identity, fmt.Errorf("given address and address recovered from signed nonce do not match")
+		return c.VerifyERC1271Signature(addrb, msgHash, signedMsg)
 	}
 
 	identity.UserID = address
 	identity.Username = address
 	return identity, nil
 }
 
+func (c *web3Connector) VerifyERC1271Signature(contractAddress common.Address, hash []byte, signedMsg string) (identity connector.Identity, err error) {
+	signature, err := hexutil.Decode(signedMsg)
+	if err != nil {
+		return identity, fmt.Errorf("could not decode hex string of signed nonce: %v", err)
+	}
+
+	if c.ethClient == nil {
+		c.logger.Errorf("Eth client was not initialized successfully %v", err)
+		return identity, fmt.Errorf("error occurred completing authentication, please try again %w", err)
+	}
+	var msgHash [32]byte
+	copy(msgHash[:], hash)
```

Comment:
Why do we need to copy?

Reply:
The `isValidSignature` requires hash of specifically `[32]byte`, the hash passed to the function isn't that.